### ISSUE TITLE
OAuth._redirectUri not using OAuth._loginStyle

### ISSUE THREAD

**Issue by saimeunt** [meteor/meteor#2758]
https://github.com/meteor/meteor/blob/devel/packages/oauth/oauth_common.js#L9

This line in OAuth._redirectUri implementation is using directly the service config document "loginStyle" property, which might be NOT defined when you use the service-configuration package directly by code (the docs use this here : http://docs.meteor.com/#meteor_loginwithexternalservice).

If you happen to define your own login system without service configuration wizard (which is always setting a valid loginStyle) and setting up service config "by hand" in the server code, you probably won't define the loginStyle property because you want Meteor to guess it by itself (using the OAuth._loginStyle logic).

Long story short, you'll end up having a bad redirect URI (with "?close" appended) and your oauth will fail.
The solution is quite simple, the code should use OAuth._loginStyle instead of tapping directly in the service config doc.
